### PR TITLE
OOP Home work 1. Reflect card flow

### PR DESCRIPTION
**Pull request by yurapukish**
Hello, happy to reach you.
Reflect card flow:
1) Add few static and class methods
2) Use getters & setters
3) Add game for you

### CODE REVIEW

File: OOP_Les1/Home.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        self._cvv = cvv
+        self.__pin = pin
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.balance = BankCard.Card_Credit_Limit
+
+    def __str__(self):
+        """Return message about card creation."""
+        return f'Created new Bank card for {self.name, self.surname}'
```
Comment (antohaUa) [inline OOP_Les1/Home.py]: Minor - > here should be not "Created" but smth aka "BankCard for ..."

File: OOP_Les1/Home.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    Card_Credit_Limit = 100
+    Exchange_Rate = {'USD': 1, 'EUR': 0.9, 'UAH': 0.03}
+
+    def __init__(self, card_num, cvv, pin, name, surname, end_date):
+        """Initialize card: add card number validation as a static method."""
+        if not BankCard.validate_card_number(card_num):
+            raise ValueError('Invalid card number')
+        self.card_num = card_num
+        self._cvv = cvv
```
Comment (antohaUa) [inline OOP_Les1/Home.py]: cvv better create as private attribute

File: OOP_Les1/Home.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        """Generate new CVV."""
+        return secrets.randbelow(900) + 100
+
+    @staticmethod
+    def validate_card_number(card_num):
+        """Validate card number."""
+        return len(card_num) == 16 and card_num.isdigit()
+
+    @classmethod
+    def get_exchange_rate(cls, rate):
```
Comment (antohaUa) [inline OOP_Les1/Home.py]: This method is not used )) Please add this method usage into withdraw logic

File: OOP_Les1/Home.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        """Initialize card: add card number validation as a static method."""
+        if not BankCard.validate_card_number(card_num):
+            raise ValueError('Invalid card number')
+        self.card_num = card_num
+        self._cvv = cvv
+        self.__pin = pin
+        self.name = name
+        self.surname = surname
+        self.end_date = end_date
+        self.balance = BankCard.Card_Credit_Limit
```
Comment (antohaUa) [inline OOP_Les1/Home.py]: Seems for balance we need setter and getter and this value better to add as key-> value param to init with default value == BankCard.Card_Credit_Limit.  Also pay attention that balance should be private or protected member - cause we should NOT allow modify it without additional checks

File: OOP_Les1/Home.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        return len(card_num) == 16 and card_num.isdigit()
+
+    @classmethod
+    def get_exchange_rate(cls, rate):
+        """Get exchange rate."""
+        return cls.Exchange_Rate[rate]
+
+    def withdraw(self, sum):
+        """Withdraw money from card."""
+        if sum > self.balance:
```
Comment (antohaUa) [inline OOP_Les1/Home.py]: Also please add here logic that payment not allowed if card.end_date expired

**Review by antohaUa — CHANGES_REQUESTED** [review]

**Comment by yurapukish** [comment]
Hello Updated code, did all changes that were requested

File: OOP_Les1/Home.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+        if sum > self.card_balance:
+            raise ValueError('Not enough money')
+        self.card_balance = (self.card_balance - sum).__round__(2)
+        print(f'Withdrew {sum}$. Balance: {self.card_balance}$')
+
+    def validate_end_date(self):
+        Card_Expired = False
+        # Determine today's date
+        date = datetime.now()
```
Comment (antohaUa) [inline OOP_Les1/Home.py:113]: For date and time comparison better to use best practice 
See examples here
https://pynative.com/python-compare-dates/

**Review by antohaUa — APPROVED** [review]

**Pull request merged**